update-doi-request/src/test/java/no/unit/nva/doi/requests/handlers/ApiUpdateDoiRequestHandlerTest.java

```diff
@@ -0,0 +1,341 @@
+    public void init() {
+        initializeDatabase();
+        Clock mockClock = Clock.fixed(mockNow, ZoneId.systemDefault());
+        doiRequestsService = new DynamoDBDoiRequestsService(client, objectMapper, environment, mockClock);
+        handler = new UpdateDoiRequestHandler(environment, doiRequestsService);
+    }
+
+    @Test
+    public void handleRequestReturnsBadRequestWhenPublicationIdIsInvalid() throws IOException {
+        var updateDoiRequest = requestWithoutPublicationId();
+        updateDoiRequest.setPublicationId(INVALID_PUBLICATION_ID);
+
+        GatewayResponse<Problem> response =  sendRequest(updateDoiRequest, USERNAME_NOT_IMPORTANT);
+
+
+        assertThat(response.getStatusCode(), is(equalTo(HttpStatus.SC_BAD_REQUEST)));
+        final Problem details = response.getBodyObject(Problem.class);
+
+        assertThat(details.getDetail(), containsString(INVALID_PUBLICATION_ID_ERROR));
+        assertThat(details.getDetail(), containsString(updateDoiRequest.getPublicationId()));
+    }
+
+    @Test
+    public void handleRequestReturnsBadRequestWhenNoPublicationId() throws IOException {
+        var updateDoiRequest = requestWithoutPublicationId();
+
+        GatewayResponse<Problem> response = sendRequest(updateDoiRequest, USERNAME_NOT_IMPORTANT,
+            Collections.emptyMap());
+
+
+        assertThat(response.getStatusCode(), is(equalTo(HttpStatus.SC_BAD_REQUEST)));
+        final Problem details = response.getBodyObject(Problem.class);
+
+        assertThat(details.getDetail(), containsString(INVALID_PUBLICATION_ID_ERROR));
+        assertThat(details.getDetail(), containsString(NULL_STRING_REPRESENTATION));
+    }
+
+    @Test
+    public void handleRequestReturnsNotFoundWhenPublicationDoesNotExist() throws IOException {
+        var updateDoiRequest = new ApiUpdateDoiRequest();
+        var notExistingPublicationIdentifier = UUID.randomUUID().toString();
+        updateDoiRequest.setPublicationId(notExistingPublicationIdentifier);
+        updateDoiRequest.setDoiRequestStatus(DoiRequestStatus.APPROVED);
+
+        GatewayResponse<Problem> response =  sendRequest(updateDoiRequest, USERNAME_NOT_IMPORTANT);
+
+        final Problem details = response.getBodyObject(Problem.class);
+
+        assertThat(response.getStatusCode(), Is.is(IsEqual.equalTo(HttpStatus.SC_NOT_FOUND)));
+
+        assertThat(details.getDetail(),
+            containsString(String.format(PUBLICATION_ID_NOT_FOUND_ERROR_FORMAT, notExistingPublicationIdentifier)));
+    }
+
+    @Test
+    public void handleRequestReturnsBadRequestWhenPublicationDoesNotHaveDoiRequest() throws IOException {
+        var publication = insertPublicationWithoutDoiRequest(
+            Clock.fixed(mockOneHourBefore, ZoneId.systemDefault()));
+
+        var updateDoiRequest = new ApiUpdateDoiRequest();
+        updateDoiRequest.setPublicationId(publication.getIdentifier().toString());
+        updateDoiRequest.setDoiRequestStatus(DoiRequestStatus.APPROVED);
```

Comment:
Are these 3 lines just a valid updateRequest or is the status that you have set important? Better to refactor to method. 
Nevertheless these 3 lines are re-occuring in many tests so it is better to refactor. Use ctrl-alt-m in IDEA for easy refactoring to method.